packages/backend/src/plugins/catalog.ts

```diff
@@ -28,12 +28,15 @@ import { useHotCleanup } from '@backstage/backend-common';
 
 export default async function createPlugin({
   logger,
-  database,
+  databaseClientFactory,
   config,
 }: PluginEnvironment) {
   const locationReader = new LocationReaders({ logger, config });
 
-  const db = await DatabaseManager.createDatabase(database, { logger });
+  const db = await DatabaseManager.createDatabase(
+    await databaseClientFactory(),
```

Comment:
Following the other suggestions, this could just be `await database.getDatabaseClient('catalog'). If we didn't want to use the `pluginId` as an argument, then maybe it's just `database.getDatabaseClient()`.

Reply:
Addressed in https://github.com/spotify/backstage/pull/2697/commits/35c0294018b8433787a795b7d9fcb6e9daef2b68.